apis/v1alpha1/generator.yaml

```diff
@@ -114,4 +114,17 @@ resources:
         - MalformedQueryString
         - MissingAction
         - MissingParameter
-        - ValidationError
\ No newline at end of file
+        - ValidationError
+    hooks:
+      delta_pre_compare:
+        code: customPreCompare(a, b)
+      sdk_create_post_set_output:
+        template_path: hooks/nodegroup/sdk_create_post_set_output.go.tpl
+      sdk_read_one_post_set_output:
+        template_path: hooks/nodegroup/sdk_read_one_post_set_output.go.tpl
+      sdk_delete_pre_build_request:
+        template_path: hooks/nodegroup/sdk_delete_pre_build_request.go.tpl
+      sdk_file_end: 
+        template_path: hooks/nodegroup/sdk_file_end.go.tpl
+    update_operation:
+      custom_method_name: customUpdate
```

Comment:
I actually find it a bit curious that you wrote a custom update method instead of using the generic hook points for `sdk_update` like you do for the create and delete code paths... any particular reason for that?

Reply:
There is no `sdkUpdate` method for node groups or cluster - since there isn't a standard "UpdateNodegroup" operation. Hence, the `sdkUpdate` is unimplemented ([see here](https://github.com/aws-controllers-k8s/eks-controller/blob/main/pkg/resource/nodegroup/sdk.go#L767)) and needs to be written from scratch.